hack/verify-crdgen.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+echo "SCRIPT_ROOT: ${SCRIPT_ROOT}"
+source "${SCRIPT_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+CRD_OPTIONS="crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Download controller-gen locally
+CONTROLLER_GEN="${GOPATH}/bin/controller-gen"
+$(go install sigs.k8s.io/controller-tools/cmd/controller-gen@latest)
+
+# Generate CRD
+api_paths="./pkg/apis/scheduling/v1alpha1/...;./vendor/github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/..."
+
+${CONTROLLER_GEN} ${CRD_OPTIONS} paths="${api_paths}" output:dir="./manifests/crds"
+
+if ! _out="$(git diff ./manifests)"; then
```

Comment:
Firstly, by default, `git diff` would always return 0, so this line would always succeed. We can add `--exit-code` to the git command to fix it.

Secondly, use `--no-pager` to ensure the diff is captured properly.
Thirdly, as we meniotned earlier, in some cases, we need to manually provide some fields, and we don't want to include those in the diff result. So we should (1) define a common pattern to detect, e.g., simply add a `# edited manually` comment on each line, and then (2) use a `-I` option to exclude the keywords.

More details, please check the suggested diff.

Reply:
Done.